Extension/src/LanguageServer/client.ts

```diff
@@ -340,11 +340,11 @@ class DefaultClient implements Client {
 
     constructor(allClients: ClientCollection, workspaceFolder?: vscode.WorkspaceFolder) {
         try {
+            this.rootFolder = workspaceFolder;
```

Comment:
Is this to resolve "workspaceFolder" first or because setting "this.rootFolder" first resolves the value?

I see you moved it up but I don't understand how this fixes the issue. If its because of ordering, please add a comment inline after it that it needs to go first to prevent it from getting reordered accidently.

Reply:
Code in in the constructor is using this.rootFolder, but it's undefined, so the ${workspaceFolderBasename} gets set to "untitled" instead. This used to work, but it must have regressed due to some other code moving around. I'm not sure what you mean by "reordered".

Reply:
Specifically, line 318 in client.ts is calling this.getName(this.rootFolder) in the Name() method, called by AdditionalEnvironment().

Reply:
I moved the setting into the creation method -- is that okay?